src/OrleansServiceFabricUtils/FabricServiceSiloResolver.cs

```diff
@@ -0,0 +1,166 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Orleans.Runtime;
+using Microsoft.Orleans.ServiceFabric.Models;
+using Microsoft.Orleans.ServiceFabric.Utilities;
+
+namespace Microsoft.Orleans.ServiceFabric
+{
+    /// <summary>
+    /// Provides information about an Orleans cluster hosted on Service Fabric.
+    /// </summary>
+    internal class FabricServiceSiloResolver : IFabricServiceSiloResolver
```

Comment:
>FabricServiceSiloResolver [](start = 19, length = 25)

Nit: this name confused me initially. But once I figured out what it is responsible for, I couldn't come up with a better name. Maybe FabricServiceMonitor or FabricClusterMonitor? But I'm not convinced they are better.